src/CustomerioTracking.tsx

```diff
@@ -143,6 +143,39 @@ class CustomerIO {
     }
     CustomerioReactnative.registerDeviceToken(token)
   }
+
+  /**
+   * Request to show prompt for push notification permissions.
+   * Prompt will only be shown if the current status is - not determined.
+   * In other cases, this function will return current status of permission.
+   * @param options
+   * @returns Success & Failure promises
+   */
+  static async showPromptForPushNotifications(options?: any) : Promise<any>{
```

Comment:
Is there a reason why `any` is used for options parameter? 

What about using a typescript `interface` instead?

Reply:
That's a good point. I used `any` to prevent customer from importing another interface or config class. But this can be changed.

Reply:
Ok, I can see how not needing to include another import could be nice. However, I believe that if a customer is using typescript in their project, they are OK with adding more imports for types in their code. Also, they probably prefer a strongly typed API when interacting with our SDK. With this in mind, I do believe it's preferred that we replace `any` with a strongly typed interface. 

If we were to organize all of the `interface`s in the project, I think the import statements for our SDK could look nice in a customer's app. 

Here is an example of the imports could look like in a customer's app: 

```ts
import type { PushPermissionOptions, Foo, Bar } from "customerio-reactnative/types" 
import { CustomerIOTracking } from "customerio-reactnative" 
...
```

----

I do not see this as a blocker for this PR.

Reply:
Small idea to share, I think changing the function name to something such as:

```suggestion
  static async askForPushPermission(options?: any) : Promise<any>{
```

would match the function naming of `getPushPermissionStatus`.

Reply:
1. This is a great suggestion. I will add this in a separate PR. I appreciate the suggestion. 


2. I used "prompt" to make it specific that calling this method would show a prompt on the screen and not do something in the background.

Reply:
1. I am OK with that while I am also OK with adding the changes to this PR. 
2. I am OK with either name. I hope using the word "prompt" doesn't confuse anyone where there are some cases where a prompt will not show up when this function is called. That's one reason why I suggested the name change.

Reply:
1. I tried implementing `customerio-reactnative/types` and found that it has to be a completely new library inside our `customerio-reactnative` with it's own package.json, index.ts and other important files. Since we are waiting to release this feature on priority, I do not want to take more time learning & implementing it at the moment. But I really loved the suggestion and will make sure to add it in the future updates. I would also like to add other configs like `CioConfigOptions` to /types so that the importing `customerio-reactive` could get reclustered as we release new features in future.

2. I still lean towards having promt in the function name as it makes things clear to me as a user. I discussed with Rehan and we both had same POV.

Reply:
1. My code sample might have been bad as the solution I am thinking shouldn't be that complex. [Here is a start to this work](https://github.com/customerio/customerio-reactnative/pull/107) that I am testing now to see if we can sneak it into this release. I think our typescript customers will appreciate the strong types. 
2. Good idea to ask others. If other people are behind it, I am for it 👍